plugin/src/test/java/com/h3xstream/findsecbugs/kotlin/KotlinHardcodedPasswordEqualsDetectorTest.java

```diff
@@ -0,0 +1,44 @@
+package com.h3xstream.findsecbugs.kotlin;
+
+import com.h3xstream.findbugs.test.BaseDetectorTest;
+import com.h3xstream.findbugs.test.EasyBugReporter;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class KotlinHardcodedPasswordEqualsDetectorTest extends BaseDetectorTest {
+
+    @Test
+    public void detectHardCodePasswordsWithEquals() throws Exception {
+        String[] files = {
+                getClassFilePath("bytecode_samples/kotlin_hardcoded_password_equals.jar"),
```

Comment:
This is perfect. Alternatively, you can write the code in Java using Kotlin API to simplify the development process.

Reply:
Good to know. I might try that in the next detectors.

Reply:
Kotlin bytecode is really close to Java bytecode. There is less difference compared to let say, Scala.